vmdb/app/controllers/provider_foreman_controller.rb

```diff
@@ -0,0 +1,535 @@
+                                                             :name   => @provider_foreman.name,
+                                                             :action => params[:id] == "new" ? "added" : "updated"})
+        replace_right_cell([:foreman_providers])
+      else
+        @provider_foreman.errors.each do |field, msg|
+          add_flash("#{field.to_s.capitalize} #{msg}", :error)
+        end
+      end
+    else
+      @provider_foreman = find_by_id_filtered(ProviderForeman, from_cid(params[:miq_grid_checks]))
+      @in_a_form = true
+      replace_right_cell
+    end
+  end
+
+  def provider_foreman_delete
+    assert_privileges("provider_foreman_delete")
+    foremen = find_checked_items
+    if foremen.empty?
+      add_flash(_("No %{model} were selected for %{task}") % {:model => ui_lookup(:tables => "providers"),
+                                                              :task  => "deletion"}, :error)
+    end
+    unless foremen.empty?
+      ProviderForeman.find_all_by_id(foremen, :order => "lower(name)").each do |foreman|
+        id = foreman.id
+        provider_name = foreman.name
+        audit = {:event        => "foreman_record_delete_initiated",
+                 :message      => "[#{provider_name}] Record delete initiated",
+                 :target_id    => id,
+                 :target_class => "Provider",
+                 :userid       => session[:userid]}
+        AuditEvent.success(audit)
+      end
+      ProviderForeman.destroy_queue(foremen)
+      add_flash(_("%{task} initiated for %{count_model} from the CFME Database") %
+                    {:task        => "Delete",
+                     :count_model => pluralize(foremen.length, "Provider")})
+    end
+    @in_a_form = false
+    @sb[:action] = nil
+    replace_right_cell
+  end
+
+  def provider_foreman_refresh
+    assert_privileges("provider_foreman_refresh")
+    @explorer = true
+    send("foreman_refresh")
+    @in_a_form = false
+    @sb[:action] = nil
+    replace_right_cell
+  end
+
+  def provider_foreman_form_fields
+    assert_privileges("provider_foreman_edit")
+    provider_foreman = find_by_id_filtered(ProviderForeman, params[:id])
+    render :json => {
+      :name       => provider_foreman.name,
+      :url        => provider_foreman.url,
+      :verify_ssl => provider_foreman.verify_ssl
+    }
+  end
+
+  def show(id = nil)
+    @flash_array = [] if params[:display]
+    @sb[:action] = params[:display]
+
+    @display = params[:display] || "main"
+    @lastaction = "show"
+    @showtype = "config"
+    @record = identify_record(id || params[:id], ConfiguredSystemForeman)
+    return if record_no_longer_exists?(@record)
+
+    @explorer = true if request.xml_http_request? # Ajax request means in explorer
+
+    @gtl_url = "/provider_foreman/show/" << @record.id.to_s << "?"
+    set_summary_pdf_data if %w(download_pdf summary_only).include?(@display)
```

Comment:
`download_pdf summary_only` seems to be a data clump as it's used down below also.

Prefer we make at minimum a constant that explains what this condition is doing.